numba/tests/test_record_dtype.py

```diff
@@ -980,5 +1009,45 @@ def test_return_charseq_tuple(self):
             self.assertEqual(expected, got)
 
 
+class TestRecordArrayGetItem(unittest.TestCase):
+    """
+    Test getitem when index is Literal[str]
+    """
+    def test_literal_variable(self):
+        arr = np.array([1, 2], dtype=recordtype2)
+        pyfunc = get_field1
+        jitfunc = njit(pyfunc)
```

Comment:
Would suggest to either use the `compile_isolated/cfunc` pattern as above or make use of the `.py_func` attr on a jit wrapped function.

Reply:
I saw the pattern above, but I didn't understand why it was being done like that. I hadn't seen that in other test modules. I used the pattern I'd seen in places like `/tests/test_array_reductions.py`. I can change it to this pattern (using compile_isolated) if it's better.

Reply:
It doesn't make a lot of difference, the `compile_isolated` is to ensure a "clean" set of typing etc. contexts when compiling. What's here is fine, it's just hat `jitfunc` has a `.py_func` attr and  I was hoping to save a few lines.

Reply:
I changed to the `compile_isolated` pattern used in other tests but I don't see how this saves a few lines (example below). I could save 1 line if I created `arr` in `setUp` and applied functions to `self.arr`. Should I do that?
```python
    def test_literal_variable(self):
        arr = np.array([1, 2], dtype=recordtype2)
        pyfunc = get_field1
        cfunc = self.get_cfunc(pyfunc, rec_dtype=recordtype2)
        self.assertEqual(pyfunc(arr[0]), cfunc(arr[0]))
```

Reply:
I was more referring to e.g. `x = njit(lambda y: y)` and `x.py_func` which would give you both a compiled and the original version. What's present in 89ed7e7 is fine, the extra line is not worth worrying about :) Thanks!